src/VisualStudio/Core/Next/Remote/JsonRpcClient.cs

```diff
@@ -127,6 +127,18 @@ private bool ReportUnlessCanceled(Exception ex, CancellationToken cancellationTo
                 return true;
             }
 
+            if (ex is OperationCanceledException && ((IDisposableObservable)_rpc).IsDisposed)
+            {
+                // JsonRpc will throw its own cancellation token if connection it is attached to
+                // is going away. we can get into this sitatuion when VS is shutting down since we dispose
```

Comment:
@AArnott can this and Hubclient.RequestService throws different exceptions such as Timeout or Connection Closed exception rather than operation cancellation exception?

when we get a dump, when exception it has is Cancellation Exception, that is really confusing what is going on.